vault/request_forwarding_rpc.go

```diff
@@ -41,12 +41,8 @@ func (s *forwardedRequestRPCServer) ForwardRequest(ctx context.Context, freq *fo
 
 	runRequest := func() {
 		defer func() {
-			// Logic here comes mostly from the Go source code
 			if err := recover(); err != nil {
-				const size = 64 << 10
-				buf := make([]byte, size)
-				buf = buf[:runtime.Stack(buf, false)]
-				s.core.logger.Error("panic serving forwarded request", "path", req.URL.Path, "error", err, "stacktrace", string(buf))
+				s.core.logger.Error("panic serving forwarded request", "path", req.URL.Path, "error", err, "stacktrace", string(debug.Stack()))
```

Comment:
Note that we way we had it is the same way http.Server does it.  It's possible they just never got around to adopting debug.Stack(), or maybe they wanted to keep a frequently firing panic from using too much memory?  Probably not, and I'm happy with the change, I just wanted to bring it up FYI.

Reply:
👍 I had a quick look out of interest, it looks like `debug.Stack` was deprecated in favour of using `runtime.Stack` between [2012](https://github.com/golang/go/commit/e49a183b760c6ac84d9fe8f63c26d92b5e162c01) and [2015](https://github.com/golang/go/commit/2d697b2401f5a840cdf4a58b414395fc729a348c), around the time that `http.Server` was still getting iterated on. Maybe that's why? I'm sure 64 KiB is enough for the vast majority of stack traces, and I'm not too worried about 64KiB being over-allocated for an event that should be so rare, so I only really changed this for consistency.

Reply:
Maybe that's it, yeah.

Should be rare, yes, but what about a panic that isn't rare, and that for some reason yields an enormous stack trace?  This seems a bit implausible to me, ignoring recursion, and if unbounded recursion is happening frequently I'm not sure the logging would tip us over the edge if we weren't there already.  In other words I'm fine with this too, I'm just explaining my reasoning why.

Reply:
@ncabatoff @tomhjp -- just a note, but PKI is probably the most likely culprit here: we suggest un-HMACing the `certificate` field to allow auditing issuance of leaf certs, but Vault likes to send full CRLs (which can be many MB) over this parameter on other endpoints depending on which field we'd hit.

It'd be great to have a way of ignoring certain (especially unauthenticated) requests or still HMACing that field on that particular request... 

But LGTM otherwise.